pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1008,17 +1029,15 @@ private CompletableFuture<? extends Subscription> getNonDurableSubscription(Stri
                 }
 
                 Position startPosition = new PositionImpl(ledgerId, entryId);
-                ManagedCursor cursor = null;
                 try {
-                    cursor = ledger.newNonDurableCursor(startPosition, subscriptionName, initialPosition,
-                            isReadCompacted);
+                    final ManagedCursor cursor = ledger.newNonDurableCursor(startPosition, subscriptionName,
+                            initialPosition, isReadCompacted);
+                    subscriptions.computeIfAbsent(subscriptionName, k ->
+                            new PersistentSubscription(this, subscriptionName, cursor, false,
+                            subscriptionProperties));
                 } catch (ManagedLedgerException e) {
                     return FutureUtil.failedFuture(e);
                 }
-
```

Comment:
Why change here?

Reply:
Just guarantee the operation `create new sub` and `unload sub` will not be executed concurrently. see https://github.com/apache/pulsar/pull/19737#discussion_r1155090950